add a new feature to remove the special tags when plugin rendering html

ColinCollins opened the pull request:
1. 增添了一个新的后缀指令 --removeTags tag1,tag2
指令会自动的去除构建文档时不需要的 tag 标记
这个指令用于指定的语言文本生成

ColinCollins commented:
1. 修改指令为 choiceLanguage 传入 zh / en : string 选定两种语言之一

ColinCollins commented:
commit2: 增添规避方案
1. 调整 json 生成方式，变为相对完整路径，解决同名文件属性覆盖问题
2. 调整 render-component，规避同名函数不同注释相互覆盖问题

ColinCollins commented:
1. 调整了最终的数据样式
2. 避开了 enum 同名数据覆盖问题
3. 后续还有待调整目前基本可用了

jareguo commented:
dist 目录干嘛要提交上来？？

jareguo commented on components/render-component.ts:
```diff
@@ -7,6 +7,12 @@ import { AttributeType } from '../utils/enums/json-keys';
+import { localize } from '../utils/helpers/localize';
+
+interface localize {
+    choiceTag: string,
```
chooseTag

jareguo commented on components/options-component.ts:
```diff
@@ -42,6 +42,11 @@ export class OptComponent extends OptionsComponent {
             help: 'Path to the json file which contains your localized template strings'
         }
 
+        const choiceLanguageOption: DeclarationOption = {
```
languageOption

jareguo commented on components/options-component.ts:
```diff
@@ -51,6 +56,8 @@ export class OptComponent extends OptionsComponent {
             includeTagsOption, 
             includeParamsOption,
             localizeOption,
```
这个 localizeOption 不就是选择语言用的吗？为什么还要加一个参数

jareguo commented on components/render-component.ts:
```diff
@@ -56,6 +76,14 @@ export class RenderComponenet extends RendererComponent {
+            // 修正 enum 重名问题的收尾修复
+            if (reflection.kind === ReflectionKind.Enum) reflection.name = reflection.name.split('_')[0];
+            // external 名称缩减，这里做简单的处理，主要是为了减少仓库维护数量。
+            if (reflection.parent && reflection.parent.name) reflection.parent.name = path.parse(reflection.parent.name.match(/[a-zA-Z0-9\/\\]/g).join('')).name;
```
没看懂这个正则操作

ColinCollins replied:
为了缩减名称，原本的 external file 是把整个路径做成 a 标签，就很长，我这边手动的缩减了。这个改法不太好，我觉得必要的话应该到 default-theme 那边去改

jareguo replied:
这个关仓库维护数量啥事？

ColinCollins replied:
emmm, 想着可以不用去 default-theme 做 external 名称显示效果的修改所以在这里顺手改了，这只是个临时方案。那我等下去掉吧

jareguo replied:
哦，你这么说我就理解了，之前一直搞不懂这个代码还能减少仓库数量，神奇

jareguo commented on components/render-component.ts:
```diff
@@ -151,12 +179,29 @@ export class RenderComponenet extends RendererComponent {
+                    reflection.comment.shortText += `\n\n${targetTags[index].text}`;
+                    targetTags.splice(index, 1);
+                    // remove language tag
+                    index = targetTags.findIndex(t => { return t.tagName === this.choiceLanguage.removeTag; });
```
如果 index === -1 咋办？

ColinCollins replied:
-1 就不管了呀，表示没有那个 tag 啊

jareguo replied:
没有这个 tag 你还 splice 干嘛？？

ColinCollins replied:
啥？ 我不是做了 index < 0 就直接下一步的判断嘛，没有里面的 splice 啊。。

jareguo commented on utils/file-operations.ts:
```diff
@@ -95,14 +95,19 @@ export class FileOperations {
-        const componentDir = splitPath[1];
-        if (fileStructureDir === "" && componentDir === undefined || parsedPath.root) {
+        const moduleDir = splitPath[1];
+        const componentDir = splitPath[2];
```
const [fileStructureDir, moduleDir, componentDir] = splitPath;

ColinCollins committed:
update own verison

Add the method to avoid engine tag and rename error 


fixed bug

1. avoid the overwrite data lost problem
